packages/across-plugin/src/service.ts

```diff
@@ -0,0 +1,530 @@
+            route.source,
+            route.destination,
+            notional
+          );
+          rates.push(quote);
+        } catch (error) {
+          console.warn(`[Across] Failed to fetch rate for route`, error);
+          // Continue with other routes even if one fails
+        }
+      }
+    }
+
+    return rates;
+  }
+
+  /**
+   * Fetch a single quote from Across API
+   */
+  private async fetchQuote(
+    source: AssetType,
+    destination: AssetType,
+    amount: string
+  ): Promise<RateType> {
+    const url = new URL(`${this.baseUrl}/suggested-fees`);
+    url.searchParams.set('inputToken', source.assetId);
+    url.searchParams.set('outputToken', destination.assetId);
+    url.searchParams.set('originChainId', source.chainId);
+    url.searchParams.set('destinationChainId', destination.chainId);
+    url.searchParams.set('amount', amount);
+
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => controller.abort(), this.timeout);
+
+    try {
+      const response = await fetch(url.toString(), {
+        method: 'GET',
+        headers: this.getHeaders(),
+        signal: controller.signal,
+      });
+
+      clearTimeout(timeoutId);
+
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${await response.text()}`);
+      }
+
+      const data: AcrossSuggestedFees = await response.json();
+
+      // Calculate effective rate with decimal normalization
+      const amountInDecimal = this.toDecimal(amount, source.decimals);
+      const amountOutDecimal = this.toDecimal(data.outputAmount, destination.decimals);
+      const effectiveRate = amountInDecimal > 0 ? amountOutDecimal / amountInDecimal : 0;
+
+      // Calculate total fees in USD (approximate based on input amount)
+      const totalFeesUsd = this.calculateTotalFeesUsd(data, amountInDecimal, source.symbol);
+
+      return {
+        source,
+        destination,
+        amountIn: amount,
+        amountOut: data.outputAmount,
+        effectiveRate,
+        totalFeesUsd,
+        quotedAt: new Date().toISOString(),
+      };
+    } finally {
+      clearTimeout(timeoutId);
+    }
+  }
+
+  /**
+   * Fetch liquidity depth at 50bps and 100bps thresholds using binary search
+   */
+  private async getLiquidityDepth(
+    routes: Array<{ source: AssetType; destination: AssetType }>
+  ): Promise<LiquidityDepthType[]> {
+    const liquidityData: LiquidityDepthType[] = [];
+
+    for (const route of routes) {
+      try {
+        // Fetch limits first to understand the range
+        const limits = await this.fetchLimits(route.source, route.destination);
+        
+        // Base amount for reference rate (1% of max)
+        const baseAmount = BigInt(limits.maxDeposit) / 100n;
+        
+        // Get reference rate with small amount
+        const baseQuote = await this.fetchQuote(route.source, route.destination, baseAmount.toString());
+        const baseRate = baseQuote.effectiveRate;
```

Comment:
We need to compare this solution deeper with the other implementation

https://github.com/Anelsonmousse/data-provider-playground/blob/7b718589429fafba3324847ee5fb1abe79caec87/packages/across-plugin/src/service.ts